### ISSUE TITLE
Second participant can't connect, grey or red screens appear on both devices

### ISSUE THREAD

**Issue by immanuelfodor** [jitsi/docker-jitsi-meet#11]
Hi, thanks for the docker-compose solution, works like a charm! :) I tried to set up the deb/manual way but this is the only one that has resulted a working video. 

However, the video feed only works if I'm the only one in the room, so it's not that fun after all :D 

- When the second participant tries to join the room on web, we both get the grey reconnect countdown screen. 
- When a mobile client tries to join a room created on web, the mobile gets the red screen, the web gets the grey screen.
- When a room is created on mobile and a second mobile tries to join, both get the red screen.

Here is my setup starting from a clean Ubuntu 18.04 LTS Cloud-Init KVM image ([https://cloud-images.ubuntu.com/bionic/current/bionic-server-cloudimg-amd64.img](https://cloud-images.ubuntu.com/bionic/current/bionic-server-cloudimg-amd64.img)) on Proxmox VE host, all commands run as `root` inside the VM:

### Installing Docker

```bash
apt update
apt upgrade -y
apt remove docker docker-engine docker.io docker-compose
apt purge docker-ce

cd ~
curl -fsSL https://get.docker.com -o get-docker.sh
sh get-docker.sh
cat /etc/apt/sources.list.d/docker.list
rm get-docker.sh 

uname -a
# Linux jitsi-meet 4.15.0-33-generic #36-Ubuntu SMP Wed Aug 15 16:00:05 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
curl -L https://github.com/docker/compose/releases/download/1.22.0/docker-compose-$(uname -s)-$(uname -m) -o /usr/local/bin/docker-compose
chmod +x /usr/local/bin/docker-compose

docker -v
# Docker version 18.06.1-ce, build e68fc7a
docker-compose version
# docker-compose version 1.22.0, build f46880fe
# docker-py version: 3.4.1
# CPython version: 3.6.6
# OpenSSL version: OpenSSL 1.1.0f  25 May 2017

systemctl status docker
# active (running)
```

### Installing Jitsi

```bash
mkdir -p /opt/jitsi-config/
chmod a+w /opt/jitsi-config/
mkdir -p ~/github
git clone https://github.com/jitsi/docker-jitsi-meet.git ~/github/docker-jitsi-meet
cd ~/github/docker-jitsi-meet
cp env.example .env
cat .env
# replacing the config values to match my environment
sed -i -E 's/(CONFIG=).*/\1\/opt\/jitsi-config/g' .env
sed -i -E 's/(TZ=).*/\1MyContinent\/MyCity/g' .env
sed -i -E "s/(JVB_COMPONENT_SECRET=).*/\1$(openssl rand -hex 32)/g" .env
sed -i -E 's/(JVB_STUN_SERVERS=).*/\1my.own.stun.server:port/g' .env
sed -i -E "s/(JICOFO_COMPONENT_SECRET=).*/\1$(openssl rand -hex 32)/g" .env
sed -i -E "s/(JICOFO_AUTH_PASSWORD=).*/\1$(openssl rand -hex 32)/g" .env
sed -i -E 's/#?(DOCKER_HOST_ADDRESS=).*/\1192.168.1.3/g' .env
cat .env

# cleanup of old configs and containers if any
docker-compose kill
docker rm docker-jitsi-meet_jvb_1 docker-jitsi-meet_jicofo_1 docker-jitsi-meet_prosody_1 docker-jitsi-meet_web_1
rm -rf ~/.jitsi-meet-cfg  
rm -rf /opt/jitsi-config/*

docker-compose up -d
docker ps
# CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS                                         NAMES
# bbc3ff069565        jitsi/jvb           "/init"             41 minutes ago      Up 39 minutes       0.0.0.0:10000->10000/udp                      docker-jitsi-meet_jvb_1
# aea6cce3f928        jitsi/jicofo        "/init"             41 minutes ago      Up 39 minutes                                                     docker-jitsi-meet_jicofo_1
# b75362a2a10b        jitsi/prosody       "/init"             41 minutes ago      Up 39 minutes       5222/tcp, 5269/tcp, 5280-5281/tcp, 5347/tcp   docker-jitsi-meet_prosody_1
# 2919f0690e92        jitsi/web           "/init"             41 minutes ago      Up 39 minutes       0.0.0.0:8000->80/tcp, 0.0.0.0:8443->443/tcp   docker-jitsi-meet_web_1
# later copypaste here than their actual start :D

chmod -R 0777 /opt/jitsi-config
```

I'm aware of this comment about storing configs outside of root-only access path, so I use a directory in `/opt` with all access provided as you can see above: https://github.com/jitsi/docker-jitsi-meet/issues/5#issuecomment-413805288
I've also found this comment about the absolute BOSH URL, tried it with and without the `DOCKER_HOST_ADDRESS` config commented but it didn't help: https://github.com/jitsi/docker-jitsi-meet/issues/2#issuecomment-409202397

Jitsi is running behind an nginx reverse proxy that is configured to `listen 443 ssl http2`, it gets valid SSL certs from LE, has a single location `/` with `proxy_pass http://192.168.1.3:8000`. Port 443/tcp is open on WAN gateway and pointed to the proxy, 10000/udp is open and pointed to the Jitsi VM (192.168.1.3:10000/udp) just as the official image says:

![Env image](https://github.com/jitsi/docker-jitsi-meet/raw/master/resources/docker-jitsi-meet.png)

Clients are latest Chrome on Linux or latest Jitsi Android apps. I must add that I've only tried to access `https://myjitsi.domain.tld` from inside the LAN on the computer but also tried to connect to it from mobile internet on the phones with same result. 

Chrome doesn't output any error on dev tools with the first participant (only complaining about a woff file but uses a fallback instead):

```
[Intervention] Slow network is detected. See https://www.chromestatus.com/feature/5636954674692096 for more details. Fallback font will be used while loading: https://myjitsi.domain.tld/fonts/jitsi.woff?94d075
F	@	jquery.min.js:2
```

When the second participant connects, an error appears for a couple of seconds until page reload:

```
[conference.js] <e.value>:  CONFERENCE FAILED: conference.videobridgeNotAvailable
```

On the page reload, there are more errors on the console:

```
Logger.js:125 [modules/xmpp/strophe.util.js] <Object.i.Strophe.log>:  Strophe: TypeError: Cannot read property 'getMemberRole' of null
    at r._onIncomingCallP2P (https://myjitsi.domain.tld/libs/lib-jitsi-meet.min.js?v=2942:2:242696)
    at r.onIncomingCall (https://myjitsi.domain.tld/libs/lib-jitsi-meet.min.js?v=2942:2:243454)
    at n.emit (https://myjitsi.domain.tld/libs/lib-jitsi-meet.min.js?v=2942:2:96313)
    at a.value (https://myjitsi.domain.tld/libs/lib-jitsi-meet.min.js?v=2942:2:738288)
    at s.Handler.run (https://myjitsi.domain.tld/libs/lib-jitsi-meet.min.js?v=2942:2:23250)
    at https://myjitsi.domain.tld/libs/lib-jitsi-meet.min.js?v=2942:2:31567
    at Object.forEachChild (https://myjitsi.domain.tld/libs/lib-jitsi-meet.min.js?v=2942:2:14907)
    at s.Connection._dataRecv (https://myjitsi.domain.tld/libs/lib-jitsi-meet.min.js?v=2942:2:31400)
    at e.Bosh._onRequestStateChange (https://myjitsi.domain.tld/libs/lib-jitsi-meet.min.js?v=2942:2:50925)
Logger.js:125 [JitsiMeetJS.js] <Object.getGlobalOnErrorHandler>:  UnhandledError: null Script: null Line: null Column: null StackTrace:  Error: Strophe: error: Cannot read property 'getMemberRole' of null
    at Object.i.Strophe.log (strophe.util.js:89)
    at Object.fatal (strophe.js:2093)
    at Object._handleError (strophe.js:2005)
    at s.Handler.run (strophe.js:2558)
    at strophe.js:3822
    at Object.forEachChild (strophe.js:1522)
    at s.Connection._dataRecv (strophe.js:3810)
    at e.Bosh._onRequestStateChange (strophe.js:5559)
Logger.js:125 [modules/xmpp/strophe.util.js] <Object.i.Strophe.log>:  Strophe: error: Cannot read property 'getMemberRole' of null
```

Then this error repeats over and over on every page reload. If one client is reloaded before another, it shows the camera image and the whole room interface but then when the second connects, grey screen again.

On mobile, there is a flash of two participants in the room (two boxes of avatars with white border) but then the red screen happens on both devices.

What am I doing wrong or is missing? It seems Jitsi is working fine when used alone but not with two participants. Do you have some recommendations what or where to investigate further?

**Comment by damencho** [comment]
Your problem is that the video bridge is not connected to the xmpp server ` CONFERENCE FAILED: conference.videobridgeNotAvailable`, you need to check the jvb logs in /var/log/jitsi/jvb.log and fix that problem.

**Comment by damencho** [comment]
Well, I'm not sure where are the log files located when running with Docker, @saghul ? `/var/log/jitsi/` is the log files location when you are deploying on a machine and not running Docker.

**Comment by immanuelfodor** [comment]
That's it, now its outputting logs like crazy, there are thousands of lines from the whole day even for timestamps when I was not connected today. What should I look for in particular, I can grep for anything that can help.

**Comment by damencho** [comment]
The problem with jvb not connected to the xmpp server, you will see it when restarting jvb.

**Comment by immanuelfodor** [comment]
There is only one error in the log right after restarting it. If these lines are not enough for you to narrow it down what I can do to fix it, I can upoad the whole output:

```
jvb_1      | JVB 2018-09-04 19:12:18.154 INFO: [10] org.jitsi.impl.configuration.ConfigurationServiceImpl.log() Normal classloader
jvb_1      | JVB 2018-09-04 19:12:18.162 INFO: [10] org.jitsi.impl.configuration.ConfigurationServiceImpl.log() failed to find jitsi-defaults.properties with class loader, will continue without it.
jvb_1      | JVB 2018-09-04 19:12:18.165 INFO: [10] org.jitsi.impl.configuration.ConfigurationServiceImpl.log() Normal classloader
jvb_1      | JVB 2018-09-04 19:12:18.168 INFO: [10] org.jitsi.impl.configuration.ConfigurationServiceImpl.log() failed to find jitsi-default-overrides.properties with class loader, will continue without it.
jvb_1      | JVB 2018-09-04 19:12:18.191 SEVERE: [10] impl.configuration.ConfigurationActivator.log() Error creating c lib instance for fixing file permissions
jvb_1      | java.nio.file.FileSystemException: /config: Operation not permitted
jvb_1      |    at sun.nio.fs.UnixException.translateToIOException(UnixException.java:91)
jvb_1      |    at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
jvb_1      |    at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
jvb_1      |    at sun.nio.fs.UnixFileAttributeViews$Posix.setMode(UnixFileAttributeViews.java:238)
jvb_1      |    at sun.nio.fs.UnixFileAttributeViews$Posix.setPermissions(UnixFileAttributeViews.java:260)
jvb_1      |    at java.nio.file.Files.setPosixFilePermissions(Files.java:2045)
jvb_1      |    at net.java.sip.communicator.impl.configuration.ConfigurationActivator.fixPermissions(ConfigurationActivator.java:166)
jvb_1      |    at net.java.sip.communicator.impl.configuration.ConfigurationActivator.start(ConfigurationActivator.java:87)
jvb_1      |    at org.jitsi.impl.osgi.framework.BundleImpl.start(BundleImpl.java:307)
jvb_1      |    at org.jitsi.impl.osgi.framework.launch.FrameworkImpl.startLevelChanged(FrameworkImpl.java:472)
jvb_1      |    at org.jitsi.impl.osgi.framework.startlevel.FrameworkStartLevelImpl$Command.run(FrameworkStartLevelImpl.java:137)
jvb_1      |    at org.jitsi.impl.osgi.framework.AsyncExecutor.runInThread(AsyncExecutor.java:122)
jvb_1      |    at org.jitsi.impl.osgi.framework.AsyncExecutor.access$000(AsyncExecutor.java:28)
jvb_1      |    at org.jitsi.impl.osgi.framework.AsyncExecutor$1.run(AsyncExecutor.java:231)
```

**Comment by damencho** [comment]
https://github.com/jitsi/docker-jitsi-meet/issues/5#issuecomment-413805288

**Comment by immanuelfodor** [comment]
Ohh, I see, even though I set the config folder to 777 on creation and after docker-compose up, it doesn't matter, as right after the up, it is still not writeable by anyone except root: 

```
# ls -la /opt/jitsi-config/                                 
total 24                                                                     
drwxrwxrwx 6 root root 4096 Sep  4 19:12 .                                   
drwxr-xr-x 3 root root 4096 Sep  4 07:16 ..                                  
drwxrwxrwx 2 root root 4096 Sep  4 19:12 jicofo                              
drwxrwxrwx 2 root root 4096 Sep  4 19:12 jvb                                 
drwxrwxrwx 4 root root 4096 Sep  4 19:12 meet                                
drwxrwxrwx 5 root root 4096 Sep  4 19:12 prosody
```

This was the current state but here comes the state after the up: 

```
# ls -la /opt/jitsi-config/        
total 24                                                                     
drwxrwxrwx 6 root root 4096 Sep  4 20:06 .                                   
drwxr-xr-x 3 root root 4096 Sep  4 07:16 ..                                  
drwxr-xr-x 2 root root 4096 Sep  4 20:06 jicofo                              
drwxr-xr-x 2 root root 4096 Sep  4 20:06 jvb                                 
drwxr-xr-x 4 root root 4096 Sep  4 20:06 meet                                
drwxr-xr-x 5 root root 4096 Sep  4 20:06 prosody
```                             

Who should be the _different user_ then, should it be added to the docker group or anything? The cloud init vm has a default user with these memberships, is it sufficient or should be altered in any way? 

```
id ubuntu                        
uid=1000(ubuntu) gid=1000(ubuntu) groups=1000(ubuntu),4(adm),20(dialout),24(cdrom),25(floppy),27(sudo),29(audio),30(dip),44(video),46(plugdev),108(lxd),114(netdev)
```

**Comment by immanuelfodor** [comment]
I tried logging in with the `ubuntu` user, then I needed to sudo the up command, and it created the config contents with `root:root`. Then I added ubuntu to the `docker` group, logged in and back and was able to run the docker commands without sudo. However, the config contents are created with root:root again. Even though the` /opt/jitsi-config` folder was set to be owned by `ubuntu:ubuntu` beforehand, the contents are recreated as root:root :(

**Comment by saghul** [comment]
What are the permissions of `/opt` and `/opt/jitsi-config` ?

**Comment by immanuelfodor** [comment]
```
drwxr-xr-x  3 root root  4096 Sep  4 07:16 opt
... 
drwxrwxrwx  2 ubuntu ubuntu 4096 Sep  4 20:41 jitsi-config
```

**Comment by immanuelfodor** [comment]
I tried setting up the whole /opt to be ubuntu:ubuntu owned and gave 777 to it, both recursively. Killed, rm-ed the containers, removed the config subfolders, up-ed the docker compose with ubuntu user, the config subfolders are still created as root:root and the error reappears in the jvb log. How can I make this work? Can I give some parameters to the docker process through the docker compose yml to which user and group to use when creating the config contents?

**Comment by immanuelfodor** [comment]
I've tried to add `user: ${CURRENT_UID}` to every block in the `docker-compose.yml` file and run the up command as `CURRENT_UID=$(id -u):$(id -g) docker-compose up -d` when acting as `ubuntu` as described here: [https://medium.com/redbubble/running-a-docker-container-as-a-non-root-user-7d2e00f8ee15](https://medium.com/redbubble/running-a-docker-container-as-a-non-root-user-7d2e00f8ee15). However, I now get these results:

```bash
docker logs docker-jitsi-meet_jvb_1
# s6-mkdir: warning: unable to mkdir /var/run/s6: Permission denied
ls -la /opt/jitsi-config/
# all 4 folders are drwxr-xr-x root:root and completely empty
docker ps
# nothing is running
```

If I don't play with the user attribute in the compose file, I get this inside the container, so I'm not sure what this "dropping privileges" mean, the user is still root inside of all four containers: 

```bash
docker exec docker-jitsi-meet_jvb_1 id
# uid=0(root) gid=0(root) groups=0(root)
```

How else can I try to overcome this permission problem and get rid of the `/config` error inside JVB and make Jitsi work? I don't have any other idea now

**Comment by saghul** [comment]
The `/config` error in the JVB is not fatal. Have you tested it again after the last changes you made?

**Comment by immanuelfodor** [comment]
Of course, after every modification I retested the whole thing with one tab open in Chrome and another in incognito mode, the same happened every time. https://meet.jit.si works this way, so I think testing this way was fair in my own instance as well.

**Comment by immanuelfodor** [comment]
Just for curiosity, now I tried it out on a laptop that never held this project.

```bash
uname -a
# Linux think-pc 4.14.69-1-MANJARO #1 SMP PREEMPT Mon Sep 10 02:18:25 UTC 2018 x86_64 GNU/Linux
# Manjaro Linux, rolling release Arch derivative

id 
# uid=1000(think) gid=1000(think) groups=1000(think),90(network),96(scanner),98(power),969(docker),986(video),988(storage),990(optical),991(lp),994(disk),995(audio),998(wheel)
# the user is member of the docker group
```

Fresh git clone, cp a new config from the example file, enter my original parameters from the Ubuntu VM (just modified the local address of my laptop), `docker-compose up -d`, then open http://127.0.0.1:8000 in a normal Chrome tab, approve webcam access, generate a new room, webcam works, copy the room link to a new incognito tab, approve webcam access, both tabs switch to the grey screen immediately. Also, all directories and files in the `/opt` config folder was created as `root:root`.

Test config for reference:

```
# Directory where all configuration will be stored.
CONFIG=/opt/jitsi-config

# Exposed HTTP port.
HTTP_PORT=8000

 # Exposed HTTPS port.
HTTPS_PORT=8443

# System time zone.
TZ=Europe/Budapest

# Internal XMPP domain. You generally don't need to change this.
XMPP_DOMAIN=meet.jitsi

# Internal XMPP domain for authenticated services. You generally don't need to
# change this.
XMPP_AUTH_DOMAIN=auth.meet.jitsi

# XMPP BOSH URL base. You generally don't need to change this.
XMPP_BOSH_URL_BASE=http://xmpp.meet.jitsi:5280

# XMPP domain for the MUC. You generally don't need to change this.
XMPP_MUC_DOMAIN=muc.meet.jitsi

# XMPP component password for Jitsi Videobridge.
JVB_COMPONENT_SECRET=xxxxxxxxxxxxxxxxx

# STUN servers used to discover the server's public IP.
JVB_STUN_SERVERS=xxxxxxxxxxxxxxxxxxxxx

# Media port for the Jitsi Videobridge
JVB_PORT=10000

# XMPP component password for Jicofo.
JICOFO_COMPONENT_SECRET=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

# XMPP user for Jicofo client connections. You generally don't need to change
# this. NOTE: this option doesn't currently work due to a bug.
JICOFO_AUTH_USER=focus

# XMPP password for Jicofo client connections.
JICOFO_AUTH_PASSWORD=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

# IP addrss of the Docker host. See the "Running on a LAN environment" section
# in the README.
DOCKER_HOST_ADDRESS=192.168.xxxxxxxxxxxxx
```

Full JVB log since I started it the first (and only) time:
[jitsi-meet-docker-manjaro-run-log.txt](https://github.com/jitsi/docker-jitsi-meet/files/2401442/jitsi-meet-docker-manjaro-run-log.txt)

I don't think it's a VM specific problem, both machines shouldn't fail the same way, there must be something we miss.

**Comment by saghul** [comment]
Can you try using https://localhost:8443 instead?

**Comment by immanuelfodor** [comment]
I also tried with setting up the default `JVB_STUN_SERVERS` from the example config, it's all the same. I was afraid that my own stun server is misconfigured or something but it seems not, it fails with Google's, too. I've also tried being on another LAN, all the same. I really wanted to make this work :(

**Comment by immanuelfodor** [comment]
Hmm, it seems https://localhost:8443/ is working fine here on the laptop just as the official meet.jit.si :O How is this possible?

**Comment by saghul** [comment]
WebRTC is only working properly on trusted (HTTPS) origins. I thought localhost was one of those... maybe not. 🤷‍♂️ 

Sorry for the delay catching this :-) I guess we can call this solved?

**Comment by immanuelfodor** [comment]
I've just tried it out on the VM, I needed to adjust the reverse proxy to the new port and https protocol in the local connection (`https->proxy->http:8000->jitsi` becomes `https->proxy->https:8443->jitsi`), and it works! :) With my own STUN server, it's not, so I need some debugging there, but with the Google ones, it's fine. I tried it with two tabs and also with one tab and a mobile, and **it works** in every case! :) 

I recommend a note in the readme to let fellow jitsi-meet-selfhosters know that they should try the https connection by default to keep them from a lot of headaches :D 

Thank you so much for the help and all the further idea to investigate. Closing this issue.
